doc/architecture/computing-services.md

```diff
@@ -0,0 +1,105 @@
+
+In the "Ocean ecosystem", Brizo is the technical component executed by the **Publishers** allowing to them to provide extended data services.
+Brizo, as part of the Publisher ecosystem, includes the credentials to interact with the infrastructure (initially cloud, but could be on-premise).
+Because of those credentials, the execution of Brizo **SHOULD NOT** be delegated to a third-party.
+
+![Brizo High-Level Architecture](img/brizo-hl-arch.png)
+
+
+## Responsibilities
+
+The main responsibilities of the software are:
+
+* Expose a HTTP api allowing to execute some extended data services (compute and/or storage)
+* Authorize the user on-chain using the proper Service Agreement. It allows to validate that user requesting the service is allowed to use that service.
+* Interact with the infrastructure (cloud/on-premise) using the Publisher credentials.
+* Start/stop/execute computing instances with the algorithm's provided by the users
+* Retrieve the logs generated during the executions
+* Register new Assets derivated of the executions as new Ocean Assets
+* Provide **Proofs of Computation** to the Ocean Network
+
+## Initial solution
+
+![Sequence Diagram for computing services](img/computing-service-flow.png)
+
+In the above diagram you can see the initial integration supported. It involves the following components/actors:
+
+* Datascientists - Final users requiring to use some computing/storage services where the data they are using is.
+* Ocean Keeper - In charge of maintain the Service Agreements and supporting the on-chain authorization
+* Publisher Infrastrucure - Cloud or On-Premise. Is were the Publisher data is.
+* Publisher Agent (Brizo) - Orchestrate/provides the additional publisher services
+* Publisher - Is the user owning the data and running Brizo. He/she doesn't interact actively in this flow.
+
+The pre-conditions of this flow are:
+
+* In a **previous stage** to this flow, the datascientist got access to an Asset (dataset) thru a classical purchase process
+* The datascientist owns an Asset of type Algorithm registered in Ocean
+* The Algorithm is implemented in one of the languages/frameworks supported by the Publisher
+* The Publisher, specified in the Asset DDO the endpoint exposed by the Publisher (Brizo) to expose the computing services
+
+
+The interactions showed in the diagram are:
+
+1. The Datascientist, after parse the DDO and see that Publisher exposes a computing service, (using Squid) call the `executeAlgorithm` method of the Publisher Brizo instance. The following parameters are given:
+   - The publicKey of the datascientist
+   - The `serviceAgreementId` related with the purchase of the Asset
+   - The `algorithmDID`
+   - Flavour of the container to execute (ie. `alpine-python:3.6`, `tensorflow:1.9`, etc.)
```

Comment:
It would be more accurate to write  

```text
- Name and tag of the Docker image to execute (e.g. `alpine-python:3.6`, `tensorflow:1.9`, etc.)
```

(The part after the colon is the "tag".)